support/util/util.go

```diff
@@ -141,3 +144,18 @@ func decompress(r io.Reader) (*bytes.Buffer, error) {
 
 	return bytes.NewBuffer(data), nil
 }
+
+// ResolveDNSHostname receives a hostname string and tries to resolve it.
+// Returns error if the host can't be resolved.
+func ResolveDNSHostname(ctx context.Context, hostName string) error {
+	timeout := time.Second
+	err := wait.PollImmediateWithContext(ctx, timeout, timeout, func(ctx context.Context) (done bool, err error) {
```

Comment:
Both `timeout` and `interval` arguments being the same means there will only be a single attempt and that `PollImmediateWithContext` in this situation is just a basic one second timeout with no retry.

Ideally, I think we should retry to decrease the condition flipping needlessly due to network latency or a single delayed/unsuccessful lookup.

However, because we are doing this synchronously in the reconcile loop, we can't really delay the reconcile for this check.

Another concerns is this condition could get stuck in the `false` state.  If, on the last reconcile before steady-state is reached, the DNS lookup fails, there is nothing that will requeue the HC and this condition could remain `false` for an indefinite time.

Reply:
So we either want to run this in a goroutine with higher timeout and retries or we fail immediately if this check fails to force a requeue?